src/Responses/TokenPaginatedResponse.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace TestMonitor\Jira\Responses;
+
+class TokenPaginatedResponse
+{
+    /**
+     * All of the items being paginated.
+     *
+     * @var array
+     */
+    protected array $items;
+
+    /**
+     * The total number of items before slicing.
+     *
+     * @var int
+     */
+    protected int $total;
+
+    /**
+     * The number of items to be shown per page.
+     *
+     * @var int
+     */
+    protected int $perPage;
+
+    /**
+     * The token for the next page.
+     *
+     * @var string
+     */
+    protected string $nextPageToken;
+
+    /**
+     * Last page.
+     *
+     * @var bool
+     */
+    protected bool $lastPage;
+
+    /**
+     * Create a new paginated response instance.
+     *
+     * @param array $items
+     * @param int $total
+     * @param int $perPage
+     * @param bool $lastPage
+     */
+    public function __construct(array $items, int $total, int $perPage, string $nextPageToken, bool $lastPage = false)
```

Comment:
lastPage is redundant, given the fact that nextPageToken is empty. The user is now enforced to provide a token.

`?string $nextPageToken = null`

And leaving out lastPage makes more sense?